obelix_client/api.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Obelix.
+# Copyright (C) 2015 CERN.
+#
+# Obelix is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Obelix is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Obelix; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Obelix-Client Search Engine."""
+
+import logging
+import re
+import time
+
+from . import utils
+
+
+CONFIG = {
+    'recommendations_impact': 0.5,
+    'score_lower_limit': 0.2,
+    'score_upper_limit': 10,
+    'score_min_limit': 10,
+    'score_min_multiply': 4,
+    'score_one_result': 1,
+    'method_switch_limit': 20,
+    'user_identifier': 'uid',
+}
+
+
+def get_logger():
+    """Get a Logger."""
+    return logging.getLogger('obelix_client')
+
+
+class Obelix(object):
+
+    """Obelix-Client."""
+
+    def __init__(self, cache_storage, recommendation_storage, queue_storage,
```

Comment:
@jirikuncar pylint is warning about to many arguments, does it make sense to do it different, if yes, how?
Thank you!

R: 50, 4: Too many arguments (6/5) (too-many-arguments)